trl/trainer/reward_trainer.py

```diff
@@ -122,11 +126,20 @@ def __init__(
                 )
             if max_length is None:
                 warnings.warn(
-                    "When using RewardDataCollatorWithPadding, you should set `max_length` in the RewardTrainer's init"
-                    " it will be set to `512` by default, but you should do it yourself in the future.",
+                    "When using RewardDataCollatorWithPadding, you should set `max_length` in RewardConfig."
+                    " It will be set to `512` by default, but you should do it yourself in the future.",
                     UserWarning,
                 )
                 max_length = 512
+            elif args.max_length is None:
+                warnings.warn(
+                    "When using RewardDataCollatorWithPadding, you should set `max_length` in RewardConfig."
+                    " It will be set to `512` by default, but you should do it yourself in the future.",
+                    UserWarning,
+                )
+                max_length = 512
+            else:
+                max_length = args.max_length
```

Comment:
I think in the case where `max_length` is not `None` then it is still overwritten when `args.max_length is None`, right? I think in that case we should keep the original value to be backwards compatible.

Reply:
Oh yes, good catch - I'll fix that

Reply:
Fixed in https://github.com/huggingface/trl/pull/726/commits/5b67e0d40d96811b3202ea5bf11628698c44b49b